### PR TITLE
FAT-1019: When patron has exceeded their Patron Group Limit for 'Maximum number of items charged out', patron is not allowed to borrow items per Conditions settings

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
## Purpose
[FAT-1019](https://issues.folio.org/browse/FAT-1019)
_When patron has exceeded their Patron Group Limit for 'Maximum number of items charged out', patron is not allowed to borrow items per Conditions settings_

## Approach
- Reproduced test scenario with UI.
- Learned necessary sent requests and then necessary received responses 
- Based on the request/response created karate test scenario

### CODE REVIEW

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature [hunk 1]
```diff
@@ -937,4 +937,110 @@ Feature: Loans tests
...
+    Given path 'patron-block-conditions'
+    When method GET
+    Then status 200
+    * def conditions = karate.sort(response.patronBlockConditions, (condition) => condition.name)
+    * def conditionId = conditions[0].id
+
+    # set block actions, renewals and requests to the condition
+    * def blockMessage = 'You have blocked!'
+    * def conditionUpdateRequest = { id: #(conditionId), blockBorrowing: false, blockRenewals: true, blockRequests: true, message: #(blockMessage), name: 'Maximum number of items charged out', valueType: 'Integer' }
+    Given path 'patron-block-conditions', conditionId
```
Comment (bisecomsergiy) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature:973]: This call we use two times. Pls, consider to move it to initData.feature

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature [hunk 2]
```diff
@@ -937,4 +937,110 @@ Feature: Loans tests
...
+    * requestEntityRequest.requestType = "Recall"
+    * requestEntityRequest.requestLevel = "Item"
+    * requestEntityRequest.instanceId = instanceId
+    * requestEntityRequest.holdingsRecordId = holdingId
+    * requestEntityRequest.pickupServicePointId = servicePointId
+    Given path 'circulation', 'requests'
+    And request requestEntityRequest
+    When method POST
+    Then status 422
+    And match $.errors[0].message == blockMessage
```
Comment (bisecomsergiy) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature:1046]: Add new line pls under this one

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature (hunk 1 above)
Comment (yusuf-murodov-epam) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature]: fixed, thanks

File: mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature (hunk 2 above)
Comment (yusuf-murodov-epam) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/loans.feature]: added

**Pull request merged**